web/client/epics/queryparams.js

```diff
@@ -86,6 +137,49 @@ const readQueryParamsOnMapEpic = (action$, store) =>
                 })
         );
 
+/**
+ * Intercept on `CLICK_ON_MAP` to get point and layer information to allow featureInfoClick.
+ * @param {external:Observable} action$ manages `CLICK_ON_MAP`
+ * @param getState
+ * @memberof epics.share
+ * @return {external:Observable}
+ */
+const onMapClickForShareEpic = (action$, { getState = () => { } }) =>
+    action$.ofType(CLICK_ON_MAP).
+        switchMap(({point, layer}) =>{
+            const allowClick = get(getState(), 'controls.share.settings.centerAndZoomEnabled');
+            return allowClick
+                ? Rx.Observable.of(resetSearch(), featureInfoClick(point, layer))
+                : Rx.Observable.empty();
+        });
+
+/**
+ * Intercept on `TOGGLE_CONTROL` to perform toggleMapInfoState and clean up on share panel close.
+ * @param {external:Observable} action$ manages `TOGGLE_CONTROL`
+ * @param getState
+ * @memberof epics.share
+ * @return {external:Observable}
+ */
+const disableGFIForShareEpic = (action$, { getState = () => { } }) =>
+    action$.ofType(TOGGLE_CONTROL).
+        switchMap((action) => {
+            if (action.control === "share") {
+                const shareEnabled = get(getState(), 'controls.share.enabled');
+                const mapInfoEnabled = get(getState(), 'mapInfo.enabled');
+                const shareParams = {bboxEnabled: false, centerAndZoomEnabled: false};
+                if (!isUndefined(shareEnabled) && shareEnabled) {
+                    return mapInfoEnabled ? Rx.Observable.of(toggleMapInfoState()) : Rx.Observable.empty();
+                }
+                return Rx.Observable.of(hideMapinfoMarker(),
+                    purgeMapInfoResults(),
+                    toggleMapInfoState(),
+                    setControlProperty("share", "settings", shareParams));
+            }
+            return Rx.Observable.empty();
```

Comment:
can you please refactor this a bit by using the filter just after actions$.ofType()
something like 

```js
action$.ofType(TOGGLE_CONTROL).filter(({control}) => control === "share").
```

and by removing the observable empty?
by doing this you avoid the execution of the epic saving a little of cpu consumption